iac-extensions/spring-config/src/main/java/org/sonar/iac/springconfig/parser/SpringConfigProfileNameUtil.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.springconfig.parser;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.sonar.iac.springconfig.tree.api.Tuple;
+
+public final class SpringConfigProfileNameUtil {
+  private static final List<String> PROFILE_NAME_PROPERTIES = List.of("spring.profiles.active", "spring.config.activate.on-profile");
+
+  private SpringConfigProfileNameUtil() {
+  }
+
+  public static String profileName(Collection<Tuple> properties) {
+    return properties.stream()
+      .filter(tuple -> PROFILE_NAME_PROPERTIES.contains(tuple.key().value().value()))
+      .filter(tuple -> tuple.value() != null)
+      .map(tuple -> tuple.value().value().value())
+      .collect(Collectors.joining(" "));
+  }
```

Comment:
I was not sure if this should be added to ProfileImpl instead of creating a new class for it.